src/map.rs

```diff
@@ -0,0 +1,54 @@
+use crate::prelude::Vec;
+use borsh::{BorshDeserialize, BorshSerialize};
+
+use crate::sdk;
+
+pub(crate) fn append_slice(id: &[u8], extra: &[u8]) -> Vec<u8> {
+    [id, extra].concat()
+}
+
+/// An non-iterable implementation of a map that stores its content directly on the trie.
+#[derive(BorshSerialize, BorshDeserialize, Default)]
+pub struct LookupMap {
+    key_prefix: Vec<u8>,
```

Comment:
Related to my comment above, you might want to make this a `KeyPrefix` type instead of arbitrary bytes.